extensions/2.0/Vendor/EXT_primitive_voxels/README.md

```diff
@@ -0,0 +1,83 @@
+# EXT_primitive_voxels
+
+<p align="center">
+  <img src="figures/voxel_cube.png">
+</p>
+
+## Contributors
+- Daniel Krupka, Cesium
+- Ian Lilley, Cesium
+- Sean Lilley, Cesium
+
+## Status
+Draft
+
+## Dependencies
+Written against the glTF 2.0 specification.
+
+## Overview
+
+This extension allows primitives to use their attribute data to represent volumetric (voxel) data.
+
+```
+{
+  "meshes": [
+    {
+      "primitives": [
+        {
+          "attributes": {
+            "_TEMPERATURE": 1
+          },
+          "mode": 2147483648,
+          "extensions": {
+            "EXT_primitive_voxels": {
+              "dimensions": [8, 8, 8],
+              "bounds": {
+                "min": [0.25, 0.5, 0.5],
+                "max": [0.375, 0.625, 0.625]
+              },
+              "padding": {
+                "before": [1, 1, 1],
+                "after": [1, 1, 1]
+              }
+            }
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+The extension adds three new primitive modes, corresponding to voxel grid geometries:
+- `0x80000000` (`2147483648`) - A Cartesian box. The grid is a Cartesian grid of equally-sized boxes.
+- `0x80000001` (`2147483649`) - A cylinder. The grid is a stack of concentric rings, evenly divided around the circumference.
+- `0x80000002` (`2147483650`) - An ellipsoid. The grid is a set of concentric ellipsoids, divided evenly in latitude and longitude.
+
+The lowest byte is reserved for future voxel modes: `0x80000000`-`0x800000FF`.
+
+|Box|Cylinder|Ellipsoid|
+| ------------- | ------------- | ------------- |
+|![Rectangular Voxel Grid](figures/box.png)|![Cylindrical Voxel Grid](figures/cylinder.png)|![Ellipsoid Voxel Grid](figures/sphere.png)|
+
+These grids all define "unit" objects centered at the origin, contained in the bounding box between `(-1, -1, -1)` and `(1, 1, 1)`. Node transforms should be used to position, orient, and scale the voxel grid as needed. The `POSITION` attribute is _not_ required or used by this extension - all positioning is through node transforms.
+
+The `dimensions` property of the extension specifies the voxel grid dimensions:
+- x/y/z for boxes
+- r/z/theta for cylinders
+- lon/lat/height for ellipsoids
+
+Dimensions must be nonzero. Elements are laid out in memory first-axis-contiguous, e.g. for boxes, `x` data is contiguous (up to stride).
```

Comment:
Perhaps that sentence can be reworded. It's row-major access where x changes the fastest. If this were a C array it would be accessed like array[z][y][x]. I wonder if array[x][y][z] is more typical...